stages/test/test_openscap_remediation.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python3
+
+import os
+import sys
+from unittest.mock import MagicMock, call, patch
+
+import pytest
+
+from osbuild import testutil
+
+STAGE_NAME = "org.osbuild.oscap.remediation"
+
+TEST_INPUT = [
+    (
+        {"datastream": "some-datastream", "profile_id": "some-profile-id"},
+        "some-datastream",
+        "some-profile-id",
+        None,
+    ),
+    (
+        {"datastream": "some-datastream", "profile_id": "some-profile-id", "tailoring": "/some/tailoring-file.xml"},
+        "some-datastream",
+        "some-profile-id",
+        "/some/tailoring-file.xml",
+    )
+]
+
+
+@pytest.fixture(name="fake_input")
+def fake_input_fixture():
+    return {
+        "type": STAGE_NAME,
+        "options": {
+            "data_dir": "/some/data/dir",
+            "config": {}
+        },
+    }
+
+
+@pytest.mark.parametrize(
+    "test_data,expected_err",
+    [
+        ({"profile_id": "some-profile-id"}, "'datastream' is a required property"),
+        ({"datastream": "some-datastream"}, "'profile_id' is a required property"),
+    ],
+)
+def test_schema_validation_oscap_remediation(fake_input, stage_schema, test_data, expected_err):
+    fake_input["options"]["config"].update(test_data)
+
+    res = stage_schema.validate(fake_input)
+
+    assert res.valid is False
+    testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+@pytest.mark.skipif(not testutil.has_executable("oscap"), reason="no oscap executable")
+@pytest.mark.parametrize("test_input,expected_datastream,expected_profile,expected_tailoring_file", TEST_INPUT)
+@patch("os.symlink")
+@patch("osbuild.util.mnt.mount")
+@patch("subprocess.run")
+def test_oscap_remediaton_smoke(
```

Comment:
There is a lot going on in this test - I wonder if that could be simplified by extracting helpers in `org.osbuild.oscap.remediation` and then testing the helpers individually where hopefully each helper would require less mocking. One possible way could be to extract a "setup_env" helper thta does the mounting and the os.symlink() and then just mock that in the integration test instead of mocking the invididal mount/os.symlink calls. Or maybe a different strategy it needs some careful looking (which I did not do) but the core of the idea would be to break `main()` into multiple functional units.

Reply:
I think this is may be a good change to make regardless of this PR. And maybe it could be extracted into a separate PR :)